### PR TITLE
fix: Added changelist variable :tada:

### PR DESCRIPTION
**Pull request by cawolfkreo**
# Description
The version number the `.hpi` file is displaying on Jenkins is `999999-SNAPSHOT`, this is because of how the `pom.xml` file is created but in theory this should override the version number.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <packaging>hpi</packaging>
 
   <properties>
-    <changelist>999999-SNAPSHOT</changelist>
```
Comment (seansaleh) [inline pom.xml]: Why also change the changelist here?
Based on the build before this change it was sufficient just to change `-Dchangelist=`
https://github.com/seansaleh/artifact-manager-s3-plugin/suites/6207879953/artifacts/219143563

File: .github/workflows/maven-automatic.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
         settings-path: ${{ github.workspace }} # location for the settings.xml file
 
     - name: Build with Maven
-      run: mvn -Drevision=1145.vb_cf6cf6ed960 -B package --file pom.xml
+      run: mvn -Dchangelist=1145.vb_cf6cf6ed960 -Drevision=1145.vb_cf6cf6ed960 -B package --file pom.xml
```
Comment (seansaleh) [inline .github/workflows/maven-automatic.yml]: This brings up the interesting question as to what version do we want the automatically built package to be. 
I would say leaving it as `999999-SNAPSHOT` is a fine choice (as upstream has it)
AKA for the `maven-automatic` job not specifying a `changelist`

Comment (seansaleh) [inline .github/workflows/maven-automatic.yml]: Do we still need `-Drevision`?

**Review by seansaleh — CHANGES_REQUESTED** [review]
Overall its great to see that we can get the version set with `-Dchangelist=`!
This PR seems to be missing the change to the manual job though

File: pom.xml (hunk 1 above)
Comment (cawolfkreo) [reply pom.xml]: For some reason I thought it didn't work and I needed to remove this. But I must've made a mistake checking the `hpi` file from a different build because the change was suficient. I'll revert this change.

File: .github/workflows/maven-automatic.yml (hunk 2 above)
Comment (cawolfkreo) [reply .github/workflows/maven-automatic.yml]: I believe some of the older branches still have the old version of the `pom.xml` file and I thought keeping the `-Drevision` flag could help with possible issues that might come from working on older branches. But since this is only going to run on the Master branch or PRs targetting Master,I think it's not necessary to keep it anymore.

TL;DR: I thought we might need it for older branches but this action only runs on master or things targetting Master so I'll remove it

Comment (cawolfkreo) [reply .github/workflows/maven-automatic.yml]: I should add, the `-Drevision` flag might be necessary for older branches if we for some reason need to do a manual release of them, so I'll keep it on the manual action.

Comment (cawolfkreo) [reply .github/workflows/maven-automatic.yml]: sounds like a good idea, I'll remove the revision and changelist flags from the automatic build and have the changelist flag on the manual release action.

**Comment by seansaleh** [comment]
This specific change looks good to me! 🚢 :shipit: 
There are a few polishing comments leftover from the previous PR at https://github.com/seansaleh/artifact-manager-s3-plugin/pull/9, but those can be done in a separate PR if you want

**Comment by cawolfkreo** [comment]
Looking at what's left from comments here and on the previous PR (https://github.com/seansaleh/artifact-manager-s3-plugin/pull/9) I think this PR is almost done except for the squash commit part. I'm thinking about squash merge this PR and then squashing the previous commits with this one on the Master branch. What do you think @seansaleh?

**Comment by seansaleh** [comment]
Squashing is good.
I'd say only bother with squashing the `master` branch once you also get in the changes from the other PR (naming and verbiage ones)
